dist/react-native/pusher.js

```diff
@@ -1702,18 +1702,18 @@ module.exports =
 	var react_native_1 = __webpack_require__(27);
 	var dispatcher_1 = __webpack_require__(14);
 	function hasOnlineConnectionState(connectionState) {
-	    return connectionState.toLowerCase() !== "none";
+	    return connectionState.type.toLowerCase() !== "none";
 	}
 	var NetInfo = (function (_super) {
 	    __extends(NetInfo, _super);
 	    function NetInfo() {
 	        var _this = this;
 	        _super.call(this);
 	        this.online = true;
-	        react_native_1.NetInfo.fetch().then(function (connectionState) {
-	            _this.online = hasOnlineConnectionState(connectionState);
+	        react_native_1.NetInfo.getConnectionInfo().then(function (connectionState) {
+	            this.online = hasOnlineConnectionState(connectionState);
```

Comment:
I'm getting a warning:

```
Possible Unhandled Promise Rejection (id: 0):
TypeError: Cannot set property 'online' of undefined
TypeError: Cannot set property 'online' of undefined
    at http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false&hot=true:76844:17
    at tryCallOne (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false&hot=true:6553:12)
    at http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false&hot=true:6639:15
    at http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false&hot=true:5958:19
    at _callTimer (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false&hot=true:5852:7)
    at _callImmediatesPass (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false&hot=true:5888:7)
    at Object.callImmediates (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false&hot=true:6092:12)
    at MessageQueue.__callImmediates (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false&hot=true:5399:16)
    at http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false&hot=true:5255:15
    at MessageQueue.__guard (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false&hot=true:5385:9)
```

The stack trace doesn't point to this obviously, but my suspect is this line. Changing `this.online` to `_this.online` fixes makes the warning go away.

Unfortunately I don't know enough about TS or have it setup to compile at all, to actually fix this properly.